### ISSUE TITLE
Error reporting

### ISSUE THREAD

**Issue by duijf** [influxdata/influxdb-java#97]
Hello,

Is there currently a way to see the results that are returned by InfluxDB? I've spent the better part of this morning trying to figure out why my data wasn't showing up in the database, and that was because I was sending a float64 value where I previously had sent an integer.

The current methods don't return much error information, so I ended up rewriting the HTTP logic myself to read the status codes.

The current methods return `void`, but if they would at least return the HTTP status code (or an exception) that would be much easier to debug when things don't work.

Are you open to a PR that implements these proposed enhancements? Or is there a way to check errors that I have overlooked?

Regards,
Laurens

**Comment by mmatloka** [comment]
+1 for this. Currently ` influxDb.write` executes without errors even when wrong server ip is defined. No info if point was sent corretly!

**Comment by mnuessler** [comment]
+1

**Comment by duijf** [comment]
Update: Currently, the company I work for (and in who's time I work with influx) decided to develop it's own system for working with InfluxDB.

Therefore I personally have nothing to gain with immediately fixing this and I can't fit a PR in my own time in the foreseable future.

I'll ask whether I can open source the package on the company's time, but our own code is a bit coupled with the rest of the system we are working on. I provide no guarantees for when, if ever I can work on this.

In short; this is up for grabs.

**Comment by jln-ho** [comment]
+1

**Comment by Rajind** [comment]
Hi all i just noticed that we can simply modify the write and query methods in influxDB interface and influxDBImpl to return Response(retrofit.client.Response) object so that we can have some insight on response. What are the cons of doing this ? Would that make affect badly for performance ?

**Comment by jln-ho** [comment]
@Rajind One con I can think of is that this approach would expose implementation details (i.e. the fact that retrofit is being used for HTTP communication) to the public API. It would probably be better to define a class ``WriteResult`` (analogous to query result) that can be constructed (package privately) from e.g. a retrofit.client.Response.

**Comment by fmachado** [comment]
I don't think this issue is valid anymore. 

```java
	/**
	 * Test for ticket #97
	 */
	@Test
	public void testTicket97() {
		String dbName = "ticket97_" + System.currentTimeMillis();
		influxDB.createDatabase(dbName);
		influxDB.setDatabase(dbName);

		Point point = Point.measurement("metric")
				.time(System.currentTimeMillis(), TimeUnit.MILLISECONDS)
				.tag("host", "host A")
				.addField("value", 5.0)
				.build();
		influxDB.write(point);

		point = Point.measurement("metric")
			.time(System.currentTimeMillis(), TimeUnit.MILLISECONDS)
			.tag("host", "host B")
			.addField("value", 5)
			.build();
		influxDB.write(point);

		influxDB.deleteDatabase(dbName);
	}
```

When I try to write to InfluxDB an integer for a field created with a float, an exception is being thrown:
```
org.influxdb.InfluxDBException: {"error":"partial write: field type conflict: input field \"value\" on measurement \"metric\" is type integer, already exists as type float dropped=1"}
	at org.influxdb.impl.InfluxDBImpl.execute(InfluxDBImpl.java:538)
	at org.influxdb.impl.InfluxDBImpl.write(InfluxDBImpl.java:312)
	at org.influxdb.impl.InfluxDBImpl.write(InfluxDBImpl.java:287)
	at org.influxdb.impl.InfluxDBImpl.write(InfluxDBImpl.java:265)
	at org.influxdb.TicketTest.testTicket97(TicketTest.java:158)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:678)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)
```

**Comment by cookiejoo** [comment]
you need to used DecimalFormat and BigDecimal
because you already know your type of value
if type is float --> must be format. (xxxx.00)

` private static final DecimalFormat df = new DecimalFormat("#.00");

    @Test
    public void write1() {
        WritePOJO writePOJO = new WritePOJO();
        writePOJO.setTimestamp(DateTime.now().getMillis());
        writePOJO.setSystemCode("xxx-app");
        writePOJO.setMeasurement("my_table");
        writePOJO.setDatabase("cat");
        Map<String, Object> fields = Maps.newHashMap();
        fields.put("failCount", 1);
        fields.put("sum", new BigDecimal(df.format(441)));
        writePOJO.setFields(fields);

        BatchWritePOJO baseMeasurement = new BatchWritePOJO();

        baseMeasurement.setSystemCode("xxx-app");
        baseMeasurement.setMeasurement("my_table");
        baseMeasurement.setDatabase("cat");
        baseMeasurement.setBatchPOJO(Lists.newArrayList(writePOJO));
        influxDbDao.batchWrite(baseMeasurement);
    }`
